src/main/java/com/gym/gymsystem/service/TraineeService.java

```diff
@@ -0,0 +1,135 @@
+        return traineeRepository.findByUser_Username(findUsername).orElseThrow(() -> new TraineeNotFoundException("Trainee not found"));
+    }
+
+    public TraineeProfileResponse getTraineeProfileAndTrainersByUsername(String findUsername) {
+        Trainee trainee = getTraineeProfileByUsername(findUsername);
+        if (trainee == null) {
+            throw new TraineeNotFoundException("Trainee not found");
+        }
+        List<Training> trainings = trainingService.getTrainingByTraineesContaining(trainee);
+
+        List<TrainerInfo> trainerInfoList = trainings.stream()
+                .flatMap(training -> training.getTrainers().stream()
+                        .map(trainer -> new TrainerInfo(trainer.getUser().getUsername(),
+                                trainer.getUser().getFirstName(),
+                                trainer.getUser().getLastName(),
+                                trainer.getSpecializations().get(0).getTrainingTypeName()))
+                )
+                .toList();
+
+        return new TraineeProfileResponse(
+                trainee.getUser().getUsername(),
+                trainee.getUser().getFirstName(),
+                trainee.getUser().getLastName(),
+                trainee.getDateOfBirth() != null ? trainee.getDateOfBirth().toString() : "N/A",
+                trainee.getAddress(),
+                trainee.getUser().getIsActive(),
+                trainerInfoList
+        );
+    }
+
+    public void updateTraineeStatus( String findUsername, boolean isActive) {
+        Trainee trainee = getTraineeProfileByUsername(findUsername);
+        if (trainee == null) {
+            throw new TraineeNotFoundException("Trainee not found");
+        }
+        trainee.getUser().setIsActive(isActive);
+        traineeRepository.save(trainee);
+    }
+
+    public Trainee updateTraineeProfile( Trainee trainee) {
+        logger.info("Updating trainee profile: {}", trainee);
+        if (trainee.getId() == null) {
+            logger.error("Trainee or Trainee ID cannot be null");
+            throw new IllegalArgumentException("Trainee or Trainee ID cannot be null");
+        }
+        userService.updateUser(trainee.getUser());
+
+        return traineeRepository.save(trainee);
+    }
+
+    @Transactional
+    public String deleteTraineeByUsername( String findUsername) {
+        Trainee trainee = traineeRepository.findByUser_Username(findUsername).orElseThrow(() -> new TraineeNotFoundException("Trainee not found"));
+
+        List<Training> trainings = trainingService.getTrainingByTraineesContaining(trainee);
+        for (Training training : trainings) {
+            training.getTrainees().remove(trainee);
+            trainingService.save(training);
+        }
+
+        traineeRepository.delete(trainee);
+        return "The Trainee has been deleted";
+    }
+
+    public Optional<Trainee> findById(Long traineeId) {
+        return traineeRepository.findById(traineeId);
+    }
```

Comment:
In this method you can check if Trainee is return from repository and return only object
see your method `getTraineeProfileByUsername` as example